opentelemetry-api/src/opentelemetry/trace/status.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import enum
+import typing
+
+
+class StatusCanonicalCode(enum.Enum):
+    """Represents the canonical set of status codes of a finished Span."""
+
+    OK = 0
+    """Not an error, returned on success."""
```

Comment:
AFAIK, attribute docstrings are not a thing in Python. So I guess to the interpreter these are just string expressions without any effect. Thus, and for consistency, I'd prefer using `#:` comments like elsewhere (https://github.com/open-telemetry/opentelemetry-python/blob/63f559ec257da7868a58699cbac598f3c57bd889/opentelemetry-api/src/opentelemetry/trace/__init__.py#L123-L139) to these "docstrings".

Reply:
To the interpreter, but not to sphinx! Sphinx [handles](https://github.com/sphinx-doc/sphinx/blob/8ea3f7fad90cf728c9c32ff2130a462a78a40515/tests/roots/test-ext-autodoc/target/enum.py) both "docstring"-style strings like this and regular comments.

I don't have a strong preference here, either comment style works for me as long as the comments and members line up in the generated docs.

@Oberon00 what's with the `:` in `#:`?